features/TESTS/filesystem/util_block_device/main.cpp

```diff
@@ -0,0 +1,171 @@
+#include "mbed.h"
```

Comment:
Great to see tests :)

Please could you: 
- add standard copyright header.
- add comment block to each test briefly explaining what they do.

Suggestion: its beneficial to generate some trace message on an assert so there is some diagnostic information in the CI if/when the assert fails.